metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -222,24 +222,47 @@ def _next_unique_table_alias(self) -> str:
         """Return the next unique table alias to use in generating queries."""
         return SequentialIdGenerator.create_next_id(StaticIdPrefix.SUB_QUERY).str_value
 
+    def _choose_time_spine_source(
+        self, agg_time_dimension_instances: Tuple[TimeDimensionInstance, ...]
+    ) -> TimeSpineSource:
+        """Determine which time spine source to use when building time spine dataset.
+
+        Will choose the time spine with the largest granularity that can be used to get the smallest granularity requested
+        in the agg time dimension instances. Example:
+        - Time spines available: SECOND, MINUTE, DAY
+        - Agg time dimension granularity needed for request: HOUR, DAY
+        --> Selected time spine: MINUTE
+        """
+        assert (
+            agg_time_dimension_instances
+        ), "Building time spine dataset requires agg_time_dimension_instances, but none were found."
+        smallest_agg_time_grain = min(dim.spec.time_granularity for dim in agg_time_dimension_instances)
+        compatible_time_spine_grains = [
+            grain for grain in self._time_spine_sources.keys() if grain.to_int() <= smallest_agg_time_grain.to_int()
+        ]
+        if not compatible_time_spine_grains:
+            raise RuntimeError(
+                # TODO: update docs link when new docs are available
+                f"No compatible time spine found. This query requires a time spine with granularity {smallest_agg_time_grain} or smaller. See docs to configure a new time spine: https://docs.getdbt.com/docs/build/metricflow-time-spine"
```

Comment:
Let's include the smallest time spine granularity as well. Something like:

`This query requires a time spine with granularity {smallest_agg_time_grain}, but the smallest available time spine granularity is {min(self._time_spine_sources.keys())}, which is too large.`